free-1.md

```diff
@@ -1,68 +1,98 @@
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Perhaps, the program can be run on terminal and at the beginning, it will list
+all the available functions. First, the user would be asked to name their diet, therefore, defining a variable, and then call different functions on the 
+variable (this can be done one by one, or in a batch mode). And when the user calls a function that indicates the end of setting up, the program will return specific instructions or meal plans for them. 
+
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+Things that are easy to do in this langauge are specifying different settings for the users' meals and, unlike any other apps of this sort, to have different settings for breakfast, lunch, and dinner, or different days of a week. 
+Something that would be difficult is to track their meals. Since there are a lot of apps that allow the user to record their meals, I feel that there's no need for this DSL to do so. This DSL's featuers will mostly focus on helping the user do something about their diet/meals beforehand, rather than afterwards. 
+
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
-aren't and conjecture why not. If so, describe them and provide links. How well 
-do they address the need? Are there any particularly admirable qualities of the
-language? Are there parts of the language you think could be improved?_
+aren't and conjecture why not. If so, describe them and provide links. How well do they address the need? Are there any particularly admirable qualities of thelanguage? Are there parts of the language you think could be improved?_
 
+One of the apps I mentioned I tried out is "myfitnesspal" (https://www.myfitnesspal.com/), which is essnetially a calorie counter. It makes it easy for the user to track what they have been eating and count calories they've been consuming. However, again, it does not let the user plan meals beforehand or evaluate how they are doing on their diet based on anything other than number of calories.
 
 ## The Project
 This section examines whether the idea makes for a good CS 111 project.
 
 
 ### Suitability
-_If someone were to work on this project, what percentage of their time would be
-spent directly engaging in the **language** aspects of this project (e.g.,
+_If someone were to work on this project, what percentage of their time would be spent directly engaging in the **language** aspects of this project (e.g.,
 making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+I believe building the systems aspects of the project would take slightly longer than the language aspects. 
+A public API for myfitnesstracker or other calorie counters might be used.
```

Comment:
Can you elaborate on your "public API"? I'm not sure how I see the DSL connecting with the terminal experience you described above. Is you DSL in API form?